Lesson4

anton2009danilov opened the pull request.

anton2009danilov committed:
Настройка BrowserRouter

anton2009danilov committed:
Передача в компонент Header параметра chatId

anton2009danilov committed:
Продолжаю по методичке.
Мой код не совпадает с методичкой.
Исправляю появившиеся баги.

anton2009danilov committed:
Починил баг

anton2009danilov committed:
Сохранение id сообщения в коллекции chats

anton2009danilov committed:
Теперь каждый чат отображает только свои сообщения

anton2009danilov committed:
Начал решать задачу №7: * Реализовать добавление новых чатов в мессенджер. Для этого нужно связать воедино чаты, хранящиеся в state MessageField, и чаты, отрисовывающиеся в ChatList.

anton2009danilov committed:
Начал работу над созданием страницы "Профиль"

anton2009danilov committed:
Сделал: можно изменить имя пользователя на странице профиля.
Новое имя отобразится в Header

anton2009danilov committed:
На странице профиля имя обновляется по нажатию "Enter"

anton2009danilov committed:
Исправил: робот подставляет "Робот" вместо имени пользователя

anton2009danilov committed:
Закончил работу на профилем: при изменении имени в профиле меняется имя в новых сообщениях

anton2009danilov committed:
Реализовал добавление нового чата через Layout state

anton2009danilov committed:
Удалил закомментированный код

anton2009danilov committed:
исправил:
MessageField.jsx:71 Uncaught TypeError: Cannot read property 'focus' of null
изменил: сообщение об изменение имени выводится через alert

anton2009danilov committed:
Сделано: при отправке пустого поля в качестве нового имени теперь назначается имя "Аноним"

anton2009danilov committed:
Исправил: пробелы в любом количестве не принимаются в качестве имени

anton2009danilov committed:
Убрал alert

SokolAlexander commented on static_src/components/ChatList.jsx:
```diff
@@ -1,29 +1,31 @@
-// import Chat from "@material-ui/icons/Chat";
+export default class ChatList extends React.Component {
+    render() {
+        const renderChats = Object.values(this.props.chats).map((chat) => {
```
лучше сделать отдельный метод

SokolAlexander commented on static_src/components/Header.jsx:
```diff
@@ -1,12 +1,44 @@
+                        <img src="../img/logo.png" alt="" className="logo" />
+                        {/* </a> */}
+                        <h3>
+                            Broken Chat{' '}
```
зачем пробел в фигурных скобках?

SokolAlexander reviewed (approved):
Все хорошо, но:
1. Организацию компонентов можно улучшить. Не используйте условный рендер в зависимости от урла внутри одного компонента - сделайте несколько компонентов и отображайте их в зависимости от урла с помощью свитч и рут. Например, если страница Профайл - показываем компонент профайл. Если чат - показываем чат. Хэдер можно использовать вне switch а имя пользователя - вынести в стейт выше по иерархии компонентов
2. Если в урл ввести например chat/11 - приложение падает. Это нужно похэндлить либо редиректом, либо показывать заглушку для несуществующих чатов.
3. На главной странице (урл /) либо сразу редиректить на chat/1, либо показывать заглушку - на данный момент по этому урлу отображается первый чат, но в адресной строке это не отражено. Урл должен соответствовать тому что отображается на странице.

anton2009danilov committed:
Работа над замечаниями к практическому заданию №4

anton2009danilov committed:
Исправил ошибки №2 и №3:
2. Если в урл ввести например chat/11 - приложение падает. Это нужно похэндлить либо редиректом, либо показывать заглушку для несуществующих чатов.

3. На главной странице (урл /) либо сразу редиректить на chat/1, либо показывать заглушку - на данный момент по этому урлу отображается первый чат, но в адресной строке это не отражено. Урл должен соответствовать тому что отображается на странице.

anton2009danilov committed:
Переделываю организацию компонентом через Router

anton2009danilov committed:
Переделал приложение согласно рекомендациям к домашнему заданию №4:

anton2009danilov committed:
Проверка является ли новое имя пробелами
